project/src/main/java/com/google/sps/servlets/PollingLocationsServlet.java

```diff
@@ -0,0 +1,86 @@
+
+@WebServlet("/pollingLocation")
+public class PollingLocationsServlet extends HttpServlet {
+  private final static String VOTER_INFO_QUERY_URL =
+    String.format("https://www.googleapis.com/civicinfo/v2/voterinfo?key=%s",
+                  Config.CIVIC_INFO_API_KEY);
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userAddress = request.getParameter("address");
+    String encodedAddress = URLEncoder.encode(userAddress);
+    String queryUrl = String.format("%s&address=%s", VOTER_INFO_QUERY_URL, encodedAddress);
+    try {
+      JsonObject voterInfoJson = queryCivicInformation(queryUrl);
+      if (voterInfoJson.has("pollingLocations")) {
+        JsonArray pollingLocationsArray = voterInfoJson.getAsJsonArray("pollingLocations");
+        JsonObject pollingLocationAddress = pollingLocationsArray.get(0).getAsJsonObject();
+        // Send data in JSON format as the servlet response for the polling location page.
+        response.setContentType("application/json;");
+        response.getWriter().println(new Gson().toJson(pollingLocationAddress));
+      }
+    } catch (IOException e) {
+      System.out.println(e);
+      return;
+    }
+  }
+
+  /** 
+   * Queries the Civic Information API and retrieves JSON response as {@code JsonObject}.
+   *
+   * @throws ClientProtocolException if the GET request to the Civic Information API fails.
+   */
+  private JsonObject queryCivicInformation(String queryUrl) throws IOException {
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    HttpGet httpGet = new HttpGet(queryUrl);
+    JsonObject json = requestHttpAndBuildJsonResponse(httpClient, httpGet);
+    httpClient.close();
+    return json;
+  }
+
+  /** 
+   * Makes HTTP GET request and converts HTTP JSON response as {@code JsonObject}.
+   *
+   * @throws ClientProtocolException if the HTTP GET request fails.
+   * @see <a href=
+   *    "https://hc.apache.org/httpcomponents-client-ga/httpclient/examples/org/apache/" +
+   *    "http/examples/client/ClientWithResponseHandler.java">Code reference</a>
+   */
+  public static JsonObject requestHttpAndBuildJsonResponse(
+      CloseableHttpClient httpClient, HttpGet httpGet) throws IOException {
+    ResponseHandler<String> responseHandler = new ResponseHandler<String>() {
+        @Override
+        public String handleResponse(final HttpResponse response) throws IOException {
+          int status = response.getStatusLine().getStatusCode();
+          if (status >= 200 && status < 300) {
+            HttpEntity entity = response.getEntity();
+            return entity != null ? EntityUtils.toString(entity) : null;
+          } else {
+            throw new ClientProtocolException("Unexpected response status: " + status);
+          }
+        }
+    };
+    String responseBody = httpClient.execute(httpGet, responseHandler);
+    return new JsonParser().parse(responseBody).getAsJsonObject();
```

Comment:
Do you have to worry about `responseBody` being null?

Reply:
Response body could be null if the API doesn't return anything. This would likely be if the user enters in an invalid address.
I'm trying to find a way to handle this edge case now.

Reply:
Would `new JsonParser().parse(null).getAsJsonObject();` throw a `NullPointerException` in that case?